src/inputs/input/input.jsx

```diff
@@ -368,10 +375,10 @@ class Input extends React.PureComponent {
                             }
                         }
                     } else {
-                        const newValueAsNumber = +newValue;
+                        let newValueAsNumber = +newValue;
 
                         if (isNumber(max)) {
-                            newValue = Math.min(max, newValueAsNumber);
+                            newValueAsNumber = Math.min(max, newValueAsNumber);
```

Comment:
Help me understand this change please.  Is it not necessary to update `newValue` as well (as the original code had it), since that is the value that will actually be used to update state?  Like maybe we should update both `newValueAsNumber` and `newValue`?

Reply:
The original implementation ignored the max ceiling. It's went like this (isNumber check omitted):
```
1. newValuesAsNumber = +newValue; // 999
2. newValue = Math.min(max, newValueAsNumber); 
// ...min(100, 999) > newValue = 100
2. newValue = Math.max(min, newValueAsNumber); 
// ...max(10, 999) > newValue = 999 (overwritten)
```
So we need to update newValueAsNumber. We could use newValue instead of newValuesAsNumber in max func, if needed.

Reply:
What if we did this?
```JavaScript
newValue = newValueAsNumber = Math.min(max, newValueAsNumber);
```